sdk/core/core-tracing/src/interfaces.ts

```diff
@@ -1,500 +1,290 @@
-/**
- * An Exception for a Span.
- */
-export declare type Exception =
-  | ExceptionWithCode
-  | ExceptionWithMessage
-  | ExceptionWithName
-  | string;
-
-/**
- * An Exception with a code.
- */
-export interface ExceptionWithCode {
-  /** The code. */
-  code: string | number;
-  /** The name. */
-  name?: string;
-  /** The message. */
-  message?: string;
-  /** The stack. */
-  stack?: string;
-}
-
-/**
- * An Exception with a message.
- */
-export interface ExceptionWithMessage {
-  /** The code. */
-  code?: string | number;
-  /** The message. */
-  message: string;
-  /** The name. */
-  name?: string;
-  /** The stack. */
-  stack?: string;
-}
-
-/**
- * An Exception with a name.
- */
-export interface ExceptionWithName {
-  /** The code. */
-  code?: string | number;
-  /** The message. */
-  message?: string;
-  /** The name. */
-  name: string;
-  /** The stack. */
-  stack?: string;
-}
-
-/**
- * Return the span if one exists
- *
- * @param context - context to get span from
- */
-export function getSpan(context: Context): Span | undefined {
-  return otTrace.getSpan(context);
-}
-
-/**
- * Set the span on a context
- *
- * @param context - context to use as parent
- * @param span - span to set active
- */
-export function setSpan(context: Context, span: Span): Context {
-  return otTrace.setSpan(context, span);
-}
-
-/**
- * Wrap span context in a NoopSpan and set as span in a new
- * context
- *
- * @param context - context to set active span on
- * @param spanContext - span context to be wrapped
- */
-export function setSpanContext(context: Context, spanContext: SpanContext): Context {
-  return otTrace.setSpanContext(context, spanContext);
+  createRequestHeaders(spanId: TracingSpanIdentifier): Record<string, string>;
 }
 
 /**
- * Get the span context of the span if it exists.
- *
- * @param context - context to get values from
+ * Options that can be passed to {@link createTracingClient}.
  */
-export function getSpanContext(context: Context): SpanContext | undefined {
-  return otTrace.getSpanContext(context);
+export interface TracingClientOptions {
+  /** The value of the az.namespace tracing attribute on any given spans */
+  namespace: string;
+  /** Information about the package invoking this trace. */
+  packageInformation: {
+    /** The name of the package. */
+    name: string;
+    /** An optional package version. */
+    version?: string;
+  };
 }
 
 /**
- * Singleton object which represents the entry point to the OpenTelemetry Context API
+ * Represents a set of items that can be set when creating a new {@link TracingContext}.
  */
-export interface ContextAPI {
-  /**
-   * Get the currently active context
-   */
-  active(): Context;
+export interface CreateTracingContextOptions {
+  /** The {@link parentContext} - the newly created context will contain all the values of the parent context unless overriden. */
+  parentContext?: TracingContext;
+  /** The span to set on the context. */
+  span?: TracingSpan;
```

Comment:
is this a parent span? what's it's relationship with `parentContext`?
Otel works with contexts rather than spans (there is more than span to pass around), we should stick to context alone.